crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -0,0 +1,674 @@
+                        message.payload_str()?,
+                        &mut self.http_proxy,
+                    )
+                    .await?)
+                }
+                Ok(MapperSubscribeTopic::ResponseTopic(ResponseTopic::SoftwareUpdateResponse)) => {
+                    debug!("Software update");
+                    Ok(
+                        publish_operation_status(message.payload_str()?, &mut self.http_proxy)
+                            .await?,
+                    )
+                }
+                Ok(MapperSubscribeTopic::ResponseTopic(ResponseTopic::RestartResponse)) => {
+                    Ok(publish_restart_operation_status(message.payload_str()?).await?)
+                }
+                Ok(MapperSubscribeTopic::C8yTopic(_)) => {
+                    debug!("Cumulocity");
+                    parse_c8y_topics(message, &self.operations, &mut self.http_proxy).await
+                }
+                _ => Err(ConversionError::UnsupportedTopic(
+                    message.topic.name.clone(),
+                )),
+            },
+        }
+    }
+
+    fn try_init_messages(&self) -> Result<Vec<Message>, ConversionError> {
+        let inventory_fragments_message = create_inventory_fragments_message(&self.device_name)?;
+        let supported_operations_message = create_supported_operations_fragments_message()?;
+        let device_data_message =
+            create_device_data_fragments(&self.device_name, &self.device_type)?;
+        let supported_log_types_message = create_supported_log_types_message()?;
+        let pending_operations_message = create_get_pending_operations_message()?;
+        let software_list_message = create_get_software_list_message()?;
+
+        Ok(vec![
+            inventory_fragments_message,
+            supported_operations_message,
+            device_data_message,
+            supported_log_types_message,
+            pending_operations_message,
+            software_list_message,
+        ])
+    }
+
+    fn sync_messages(&mut self) -> Vec<Message> {
+        let sync_messages: Vec<Message> = self.alarm_converter.sync();
+        self.alarm_converter = AlarmConverter::Synced;
+        sync_messages
+    }
+}
+
+async fn parse_c8y_topics(
+    message: &Message,
+    operations: &Operations,
+    http_proxy: &mut impl C8YHttpProxy,
+) -> Result<Vec<Message>, ConversionError> {
+    match process_smartrest(message.payload_str()?, operations, http_proxy).await {
+        Err(
+            ref err @ CumulocityMapperError::FromSmartRestDeserializer(
+                SmartRestDeserializerError::InvalidParameter { ref operation, .. },
+            ),
+        ) => {
+            let topic = C8yTopic::SmartRestResponse.to_topic()?;
+            let msg1 = Message::new(&topic, format!("501,{operation}"));
+            let msg2 = Message::new(&topic, format!("502,{operation},\"{}\"", &err.to_string()));
+            error!("{err}");
+            Ok(vec![msg1, msg2])
```

Comment:
Why do we send in a single batch both `501 Executing` and `502 Failed` ?

I got it looking for the place where `503 Success` are sent! But this is not obvious. I would add a comment to say that sending the `501`, `502` and `503` messages are the responsibility of the operation plugin ... unless one cannot parse the command.